### ISSUE TITLE
Complete experiment with previous commit with results

### ISSUE THREAD

**Issue by zliuar** [Rayhane-mamah/Tacotron-2#164]
Hi, I did an experiment with the model of the commit right before Aug 4. And now I had some results.

The Tacotron training is done with 20k steps.
The WaveNet training seems converge after 800k steps with training loss around 6.
<img width="1504" alt="training" src="https://user-images.githubusercontent.com/22195768/44398505-9dfb0d80-a576-11e8-8e50-e1adafcca8cb.png">

And here is the output.
[GL_output.zip](https://github.com/Rayhane-mamah/Tacotron-2/files/2306023/GL_output.zip)
[900k.zip](https://github.com/Rayhane-mamah/Tacotron-2/files/2306024/900k.zip)

You may hear that the quality of this wavenet generated audio is not as good as the griffin lim output. Also it is not as good as the speech in the r9y9's post [here](https://r9y9.github.io/wavenet_vocoder/).

@Rayhane-mamah Do you think should there be some problems with my experiments? Is the training loss around 6 normal? The settings are the default settings (raw mode). And I am also confused why the speech of the first testing sentence in that post has 3 sec duration while I have 4 sec duration? Any suggestions to control the speech duration?

Thanks!

**Comment by zliuar** [comment]
The text for synthesized speech are the following.
32 'How are you?'
33 'What is going on?'
34 'Nice to meet you!'
35 'I went to school.'
36 'I love food.'
37 'attention based neural network'

**Comment by Rayhane-mamah** [comment]
Hi @zliuar, MoL usually takes more than 1 million steps to give good results. r9y9 in his experiments made 4 trainings of 300k each where each time he resets the learning rate and EMA variables. So I guess it still needs tuning?

**Comment by zliuar** [comment]
@Rayhane-mamah Thanks for your reply! I have seen your replies in other issues about the steps for wavenet to converge. But I feel confused since my loss graph seems converged.

How much do you suggest to reduce the learning rate each time? And how about the ema? I am not very experienced on tuning these params. Could you help me with it? Thank you very much!

**Comment by Rayhane-mamah** [comment]
@zliuar you're most welcome :)

Actually, for wavenet I haven't quite figured out the best parameters for training yet.. it's tough to train models that generate raw audio samples really.. What I noticed, is that with 1e-4 sometimes the model gets stuck in a bad local minimum, which we don't like. A possible solution for this is to use what r9y9 does. Rerun training by manually setting a "bump point" for the learning rate to act as if it's a fresh start.

For example you may if you have trained for 900k. you may change the global step during optimizer creation to be global_step - 900k or something similar. of course making a backup checkpoint is a good idea at this point :) 

I am really wondering if noam is the best learning rate scheme at this point, but I'm still doing some tests here and there, I can't quite give any good opinions before testing stability and repeatability. One thing is certain for now, repeating wavenet results is very tough with our current parameters.

**Comment by zliuar** [comment]
@Rayhane-mamah Wow, I didn't realize the complexity of training a wavenet model. Thanks for your explanation! I will follow your advice and let me see what results could be obtained.

**Comment by Rayhane-mamah** [comment]
Yeah I didn't think it would be like that too :) @Yeongtae is having fastest converging model for now I think. I on my side am trying different learning rate schemes and optimisation parameters (also experimenting with loss formulation) to try getting the most stable, fastest to converge model. I'll keep you informed of our advance ;)

**Comment by zliuar** [comment]
@Rayhane-mamah Thank you very much!

**Comment by atreyas313** [comment]
Hi @Rayhane-mamah, 
The synthesis output to step 61K is as follows:
[wavenet_output.zip](https://github.com/Rayhane-mamah/Tacotron-2/files/2315749/wavenet_output.zip)


But the result is very bad, do you think this is so normal?
Is it for you possible to provide samples of LJspeech?
